draft-ietf-jsonpath-base.md

```diff
@@ -1467,44 +1467,30 @@ a type system is first introduced.
 
 Each parameter as well as the result of a function extension must have a declared type.
 
-A type is a set of instances.
 Declared types enable checking a JSONPath query for well-typedness
 independent of any argument the JSONPath query is applied to.
 
-{{tbl-types}} defines the available types in terms of abstract instances, where `v` denotes a value, and `nl` denotes
-a nodelist.
+{{tbl-types}} defines the available types in terms of the instances they contain.
 
-| Type                 | Abstract Instances                       |
-| :--                  | :----------------                        |
-| `ValueType`          | `Value(v)`, `Nothing`                    |
-| `LogicalType`        | `LogicalTrue`, `LogicalFalse`            |
-| `NodesType`       | `Nodes(nl)`                              |
+| Type                 | Instances                       |
+| :--                  | :------------------------------ |
+| `ValueType`          | JSON values or `Nothing`        |
+| `LogicalType`        | `LogicalTrue` or `LogicalFalse` |
+| `NodesType`          | Nodelists                       |
 {: #tbl-types title="Function extension type system"}
 
 Notes:
 
-* `ValueType` is an abstraction of a JSON value or `Nothing`.
-* `LogicalType` is an abstraction of the result of a `logical-expr`.
-  Its two instances, `LogicalTrue` and `LogicalFalse`, are not related to
-  the JSON literals `true` and `false` and have no direct syntactical representation in JSONPath.
-* `NodesType` is an abstraction of a `filter-query` (which appears
-  in a test expression or as a function argument).
-  Members of `NodesType` have no direct syntactical representation in JSONPath.
-
-The abstract instances above can be obtained from the concrete representations in {{tbl-typerep}}.
-
-| Abstract Instance | Concrete Representations                                           |
-| :---------------: | :----------------------:                                           |
-| `Value(v)`        | JSON value `v`                                                     |
-| `Nothing`         | A representation of the absence of a JSON value, distinct from the JSON literal `null`, e.g., from a Singular Query or `filter-query` resulting in an empty nodelist   |
-| `Nodes(nl)`       | A list of zero or more nodes, e.g., from a `filter-query` resulting in the nodelist `nl`, which may or may not be empty  |
-{: #tbl-typerep title="Concrete representations of abstract instances"}
+* The only instances that can be directly represented in JSONPath syntax are the literal and
+  primitive values of `ValueType`.
```

Comment:
I like that. @cabo and @gregsdennis: WDYT?

Reply:
I don't think we should have three two words together since they describe two different sets of values.

I would be happy with "literals and primitives" or "literals and primitives values."

Also...
![Screenshot_20230316-224716.jpg](https://user-images.githubusercontent.com/2676804/225579644-9d67181e-1d5e-40da-8d17-d47190058673.jpg)

Reply:
All of our literals express primitive values, so this is indeed a pleonasm ("doppeltgemoppelt").
It probably cannot hurt to mention this property, but not in a way that it seems that we are saying there are literals that are expressing non-primitive (structured) values.

Reply:
JSON was started by Mr. Crockford, taking a subset of JavaScript literals - especially object and array literals - for the definition of a data exchange format ... that's the history ... at least as I am aware of.

So in JSON context non-primitive (object and array) literals are well known, thus in contrast 'primitive literals' makes perfect sense.

Reply:
All of our literals express primitive values, so it is confusing to occasionally qualify "literals" with "primitive".
(It is also slightly wrong, but acceptably so.)

Reply:
I've attempted a rewording.